lbz/events/api.py

```diff
@@ -52,9 +56,22 @@ def set_resources(self, resources: List[str]) -> None:
     def set_bus_name(self, bus_name: str) -> None:
         self._bus_name = bus_name
 
+    @property
+    def sent_events(self) -> List[BaseEvent]:
+        return deepcopy(self._sent_events)
+
+    @property
+    def pending_events(self) -> List[BaseEvent]:
+        return deepcopy(self._pending_events)
+
+    @property
+    def failed_events(self) -> List[BaseEvent]:
+        return deepcopy(self._failed_events)
+
     def register(self, new_event: BaseEvent) -> None:
         self._pending_events.append(new_event)
 
+    # TODO: Stop sharing protected lists outside the class, use the above properties instead
```

Comment:
_FYI:_ I have added above the properties that always return a copy of events, just to do it the right way.

If you think, we can remove the below methods, I can do this but I left them for backward compatibility 😉